### PR TITLE
Added: Polygon and Vec2 typings

### PR DESCRIPTION
**Pull request by Nielio**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by Nielio** [comment]
@tmpvar FYI: Just created typings for two of your repos

File: types/polygon/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+    /**
+     * Ensure all of the points are unique
+     * @param returnNew If passed a truthy value, a new Polygon will be returned to you. Otherwise the operation will be applied to this and this will be returned.
+     */
+    dedupe(returnNew?: boolean): Polygon;
+
+    /**
+     * Insert `vec2` at the specified index
+     * @param vec2
```
Comment (SimonSchick) [inline types/polygon/index.d.ts]: Nit: You can ditch all the doc tags with no descriptions.

File: types/polygon/polygon-tests.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// no tests yet
+const ok = 1;
```
Comment (SimonSchick) [inline types/polygon/polygon-tests.ts]: Time to write some!

File: types/vec2/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ * A generic library useful when you need to work with points/vectors in 2d space.
+ * **Stuff to Note**: most of the Vec2's methods take a `returnNew` as the last parameter.  If passed a truthy value, a new vector will be returned to you.
+ * Otherwise the operation will be applied to `this` and `this` will be returned.
+ * Also, since `Infinity`and `NaN` are so insidious, this library will throw as soon as it detects either of these so you can take action to fix your data/algorithm.
+ */
+declare class Vec2 {
+    x: number;
```
Comment (SimonSchick) [inline types/vec2/index.d.ts]: `public readonly`?

**Review by SimonSchick — CHANGES_REQUESTED** [review]

File: types/polygon/index.d.ts (hunk 1 above)
Comment (Nielio) [reply types/polygon/index.d.ts]: I dropped them

File: types/vec2/index.d.ts (hunk 3 above)
Comment (Nielio) [reply types/vec2/index.d.ts]: The library allows to write directly into theese attributes.
But since, there is a function `change()` witch sets the values, i decided to hint them as `readonly`, though.
Also i added the `public` keyword everywhere.

File: types/polygon/polygon-tests.ts (hunk 2 above)
Comment (Nielio) [reply types/polygon/polygon-tests.ts]: I wrote some very simple tests to check the instance creation and the way the modules are imported. Is this ok?
I am not the author of the original library. Just wanted to provide typings for it.

**Review by SimonSchick — APPROVED** [review]

File: types/polygon/tslint.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "member-access": false
```
Comment (sandersn) [inline types/polygon/tslint.json]: Why is this needed? Is it possible to remove it?

File: types/vec2/tslint.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "member-access": false
```
Comment (sandersn) [inline types/vec2/tslint.json]: Same here.

File: types/vec2/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+    public toJSON(): { x: number, y: number };
+
+    /**
+     * _returns_: `'(x, y)'`
+     */
+    public toString(): string;
+}
+
+export default Vec2;
```
Comment (sandersn) [inline types/vec2/index.d.ts]: Looking at 
https://unpkg.com/vec2@1.6.0/vec2.js, it looks like the shipped version has no default property on module.exports.  So it should be `export = Vec2`. 

Probably true of polygon too.

**Review by sandersn — CHANGES_REQUESTED** [review]
Polygon doesn’t have a default export either.

File: types/vec2/index.d.ts (hunk 6 above)
Comment (Nielio) [reply types/vec2/index.d.ts]: Ok. I changed it like you have proposed.

But for polygon i have to write `export default Polygon`.
If i use `export = Polygon`, i get the error `Module '.../DefinitelyTyped/types/polygon/index"' has no default export.` running `npm test`

File: types/polygon/tslint.json (hunk 4 above)
Comment (Nielio) [reply types/polygon/tslint.json]: I added this, to be able to write `public` on functions and properties, without `public' is implicit` warning.
Now i removed `public` and the rule. Hope this is ok.

File: types/vec2/index.d.ts (hunk 6 above)
Comment (sandersn) [reply types/vec2/index.d.ts]: well, I think the emitted JS code will be `var Vec2 = require("vec2")`, so the equivalent Typescript import should be `import Vec2 = require("vec2")` not `import Vec2 from "vec2"`.

Comment (Nielio) [reply types/vec2/index.d.ts]: Yeah, this is exactly what i tried and got the mentioned errors. See this [TravisRun](https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/507749764?utm_source=github_status&utm_medium=notification)
I applied your proposal for vec2 and polygon. For vec2 only, the error is also present.

(Sorry for my late response, was very sick last week)

Comment (sandersn) [reply types/vec2/index.d.ts]: I don't think the PR ever had all four of the import/exports in the following state, so the build was always reporting mismatches:

```
export = Vec2
import vec2 = require("vec2")

export = Polygon
import polygon = require("polygon")
```

(I guess it is five actually, since polygon depends on vec2 and has an import statement for it.)

If they all match, that should resolve the errors.

Comment (Nielio) [reply types/vec2/index.d.ts]: Ouh. My bad. Forgot the correct import in the test files 🙈 
Now everything is working!

**Review by sandersn — APPROVED** [review]
